File: Expense_Tracker/Expense_Tracker_Backend/expense_tracker_backend/src/main/java/com/example/demo/service/TransactionService.java
package com.example.demo.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.example.demo.model.Transaction;
import com.example.demo.repository.TransactionRepository;

@Service
public class TransactionService {
	
	@Autowired
	private TransactionRepository repository;
	
	public List<Transaction> getTransactions()
	{
		return repository.findAll();
	}
	
	public Transaction getTransactionbyid(Long id)
	{
		return repository.findById(id).get();
	}
	
	public Transaction createTransaction(Transaction transaction)
	{
		return repository.save(transaction);
	}
	
	public Transaction updateTransaction(Long id, Transaction new_transaction)
	{
		Transaction old_transaction=repository.findById(id).get();
		old_transaction.setAmount(new_transaction.getAmount());
		old_transaction.setDescription(new_transaction.getDescription());
		old_transaction.setType_of_transaction(new_transaction.getDescription());
		return repository.save(old_transaction);
	}
	
	public String deleteTransaction(Long id)
	{
		repository.deleteById(id);
		return "Trasaction "+id+" deleted";
	}

}

File: Expense_Tracker/Expense_Tracker_Frontend/src/services/TransactionService.js
import axios from 'axios'

const EMPLOYEE_BASE_REST_API_URL = 'http://localhost:8081/api/v1/';

class TransactionService
{
    getTransactions()
    {
        return axios.get(EMPLOYEE_BASE_REST_API_URL)
    }

    getTransactionbyid(transactionId)
    {
        return axios.get(EMPLOYEE_BASE_REST_API_URL+ 'gettransaction/'+transactionId)
    }

    createTransaction(transaction)
    {
        return axios.post(EMPLOYEE_BASE_REST_API_URL+'createTransaction',transaction)
    }

    updateTransaction(transactionId,transaction)
    {
        return axios.put(EMPLOYEE_BASE_REST_API_URL+'updatetransaction/'+transactionId,transaction)
    }

    deleteTransaction(transactionId)
    {
        return axios.delete(EMPLOYEE_BASE_REST_API_URL+'deleteTransaction/'+transactionId)
    }
}

export default new TransactionService